modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/AbstractStringProcessor.java

```diff
@@ -29,10 +29,8 @@
 /**
  * Base class for processors that manipulate source strings and require a single "fields" array config value, which
  * holds a list of field names in string format.
- *
- * @param <T> The resultant type for the target field
  */
-abstract class AbstractStringProcessor<T> extends AbstractProcessor {
```

Comment:
There's  more classes to change, but I'm wondering if ti's worth changing the classes that extend from this  to include a type instead like so:
```
public final class BytesProcessor extends AbstractStringProcessor<Long> {
```

This would allow us to keep type safety for the process issue

Reply:
I can go either way. The type wasn't really used, so I thought I should remove it, but what you;re suggestion is good too, as it remove those -Xlint exclusions too. @jakelandis what do you think?

Reply:
I would prefer to leave the generic type there.  I think I added it, but didn't bother with explicitly satisfying the generic since is implicitly satisfied by the `abstract T process` method. 

It looks like the lint exclusions pre-dated this change, and not sure if explicitly satisfying the generic will allow for a clean removal of the exclusion. (I am bit surprised that implicitly satisfying generics trigger lint warnings)

Reply:
I pushed: c3746d8dadf5f4533fa0d39a4ea119443bcbc3a3